### DISCUSSION TITLE
OpenID Connect configured to use a third party Authentication client - how can I use the access token issued by them to access their API

### DISCUSSION THREAD

**Discussion by stebrough** [OrchardCMS/OrchardCore#15220] [Q&A]
Hi all,

I've got Orchard OpenID Connect configured to use a third party Authentication client. All seems to be setup well, I can authenticate properly, and I can also see Orchard getting a token from them successfully. Here is where I'm stuck, if I then want to query the third party API using that token, how do I go about doing this? 

Thanks!

**Comment by ns8482e** [comment]
If I am not mistaken, there is option to save external tokens in openid settings, also same can be configured using configuration

**Reply by stebrough** [reply]
Hiya, yes I have that option checked. I just have no idea how to use that token to call the other API

**Reply by ns8482e** [reply]
You can use the token to access third api that is protected by the same authorization server that generated the token and you have valid scope to access that third party api

**Reply by stebrough** [reply]
When you say valid scope, do you mean the scope passed to the third party, or the scope should be what the third party api is set as?
